lib/app/app.dart

```diff
@@ -1,24 +1,55 @@
+import 'package:first/provider/theme_provider.dart';
 import 'package:flutter/material.dart';
+import 'package:provider/provider.dart';
 import './modCalculator.dart';
 
 class App extends StatelessWidget {
-  const App({Key? key}) : super(key: key);
-
   @override
   Widget build(BuildContext context) {
-    return MaterialApp(
-      title: 'Modder',
-      debugShowCheckedModeBanner: false,
-      theme: ThemeData(
-        primarySwatch: Colors.blue,
-      ),
-      home: Scaffold(
-        appBar: AppBar(
-          title: Text('Modder'),
-          centerTitle: true,
-        ),
-        body: ModCalculator(),
+    return Scaffold(
+      appBar: AppBar(
+        title: const Text('Modder'),
+        centerTitle: true,
+        actions: [
+          PopupMenuButton(
+            itemBuilder: (context) => [
+              PopupMenuItem(
+                enabled: false,
+                child: ListTile(
+                  title: Padding(
+                    padding: const EdgeInsets.only(left: 16),
+                    child: Text(
+                      'Dark Mode',
+                      style: TextStyle(
+                        fontWeight: FontWeight.w600,
+                      ),
+                    ),
+                  ),
+                  trailing: buildSwitch(),
+                ),
+              ),
+            ],
+          ),
+        ],
       ),
+      body: ModCalculator(),
+    );
+  }
+
+  StatefulBuilder buildSwitch() {
+    return StatefulBuilder(
+      builder: (BuildContext context, StateSetter setState) {
+        return Switch(
+          value: context.read<ThemeProvider>().isDarkMode,
+          // value: Provider.of<ThemeProvider>(context).isDarkMode,
```

Comment:
Remove if not needed

Reply:
that is for learning purpose. Provider package still has backward compatibility and the comment version also works which is much more easier to understand and learn. Tei timi harulai bujhna huncha vanera rakheko tyo provider ko chai

Reply:
Okay :)